backend/effects/builtin/fileWriter.js

```diff
@@ -40,22 +37,30 @@ const fileWriter = {
             <file-chooser model="effect.filepath" options="{ filters: [ {name:'Text',extensions:['txt']} ]}"></file-chooser>
         </eos-container>
 
-        <eos-container header="Text" pad-top="true">
-            <input ng-model="effect.text" type="text" class="form-control" id="chat-text-setting" placeholder="Enter text" replace-variables>
+        <eos-container header="Write Mode" pad-top="true">
+            <div class="controls-fb-inline" style="padding-bottom: 5px;">
+                <label class="control-fb control--radio">Replace
+                    <input type="radio" ng-model="effect.writeMode" value="replace"/> 
+                    <div class="control__indicator"></div>
+                </label>
+                <label class="control-fb control--radio">Append
+                    <input type="radio" ng-model="effect.writeMode" value="append"/>
+                    <div class="control__indicator"></div>
+                </label>
+                <label class="control-fb control--radio">Delete Line
+                    <input type="radio" ng-model="effect.writeMode" value="delete"/>
+                    <div class="control__indicator"></div>
+                </label>
+                <label class="control-fb control--radio">Delete All
+                    <input type="radio" ng-model="effect.writeMode" value="delete-all"/>
+                    <div class="control__indicator"></div>
+                </label>
+            </div>
         </eos-container>
 
-        <eos-container header="Write Mode" pad-top="true">
-        <div class="controls-fb-inline" style="padding-bottom: 5px;">
-            <label class="control-fb control--radio">Replace
-                <input type="radio" ng-model="effect.writeMode" value="replace"/> 
-                <div class="control__indicator"></div>
-            </label>
-            <label class="control-fb control--radio">Append
-                <input type="radio" ng-model="effect.writeMode" value="append"/>
-                <div class="control__indicator"></div>
-            </label>
-        </div>
-    </eos-container>
+        <eos-container header="Text or Line Number(s)" pad-top="true" ng-if="effect.writeMode === 'replace' || effect.writeMode === 'append' || effect.writeMode === 'delete'">
+            <input ng-model="effect.text" type="text" class="form-control" id="chat-text-setting" placeholder="Enter text" replace-variables>
+        </eos-container>
```

Comment:
Hopefully its not too nit-picky, but I'd like to try to avoid having one text field be used for two completely different purposes. It makes things a little ambiguous for the user but very ambiguous in the code. Having separate text fields would also let us only show variables that only output numbers in the Line Number(s) field.

Proposed change:
```html
<eos-container header="Text" pad-top="true" ng-if="effect.writeMode === 'replace' || effect.writeMode === 'append'">
    <input ng-model="effect.text" type="text" class="form-control" id="chat-text-setting" placeholder="Enter text" replace-variables>
</eos-container>

<eos-container header="Line Number(s)" pad-top="true" ng-if="effect.writeMode === 'delete'">
    <p class="muted">Enter a line number or list of line numbers (separated by commas) to delete.</p>
    <input ng-model="effect.lineNumbers" type="text" class="form-control" id="chat-line-numbers-setting" placeholder="Enter line number(s)" replace-variables="number">
</eos-container>
```
Note that I did` replace-variables="number"`. This will make the variables menu only show variables that have an output type of "number". 

Also note that you will have to update `fileWriterProcessor.js` to account for the now separated `lineNumbers` property.